### PR TITLE
Finish todo app

### PR DESCRIPTION
**Pull request by preciousuwhubetine**
In this branch, I built the complete todo app. I added the following features:
- Listing to-do tasks.
- Checking off tasks as completed.
- Editing to-do tasks.
- Deleting to-do tasks.
- User authentication.
- Responsive navigation bar.
- User profile page.

### CODE REVIEW

File: src/context/TodosContext.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  };
+
+  const setUpdate = (updatedTitle, id) => {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.id === id) {
+          todo.title = updatedTitle;
+        }
+        return todo;
+      }),
```
Comment (adamilare) [inline src/context/TodosContext.jsx]: [Optional] To ensure `no-param-reassign` ESLint check passes, you can do the following;
```suggestion
      todos.map((todo) => {
        if (todo.id === id) {
          return {...todo, title: updatedTitle};
        }
        return todo;
      }),
```

File: src/components/TodoItem.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        {user && (
+          <button type="button" onClick={handleEditing}>
+            <AiFillEdit
+              style={{ color: '#5e5e5e', fontSize: '16px' }}
+            />
+          </button>
+        )}
+        <button type="button" onClick={() => delTodo(itemProp.id)}>
+          <FaTrash style={{ color: '#5e5e5e', fontSize: '16px' }} />
+        </button>
```
Comment (adamilare) [inline src/components/TodoItem.jsx:59]: [Optional] `delete' is part of edit operations, so supposed to be included in the logic;
```suggestion
        {user && (
          <button type="button" onClick={handleEditing}>
            <AiFillEdit
              style={{ color: '#5e5e5e', fontSize: '16px' }}
            />
          </button>
        )
        <button type="button" onClick={() => delTodo(itemProp.id)}>
          <FaTrash style={{ color: '#5e5e5e', fontSize: '16px' }} />
        </button>}
```

File: src/components/ProtectedRoute.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable react/prop-types */
```
Comment (adamilare) [inline src/components/ProtectedRoute.jsx]: [Optional] You might consider implementing props validation, as this is essential to catch bugs during development

File: src/components/Navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                </li>
+              )}
+            </React.Fragment>
+          ))}
+          {user && (
+            <div className="logout">
+              <p>{user}</p>
+              <button type="button" onClick={handleLogout}>Logout</button>
+            </div>
+          )}
```
Comment (adamilare) [inline src/components/Navbar.jsx]: This block supposed to be after `</nav>` for better UI/UX and to keep to the provided template.

**Your implementation**
Navbar user position

**Expected**
Navbar user position 2

**Review by adamilare — CHANGES_REQUESTED** [review]
Hi @preciousbetine 👋 , 

# ♻️ Changes Required ♻️

so-close-jim-halpert

Good job so far!
There are some little changes that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Readme descriptive
✔️ Commit messages meaningful and descriptive
✔️ PR summary present
✔️ Linter checks are green
✔️ App runs successfully
✔️ Correct use of Git flow 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/TodoItem.jsx (hunk 2 above)
Comment (preciousuwhubetine) [reply src/components/TodoItem.jsx:59]: @adamilare I think the login is meant to grant users access to the edit function alone. A task can be deleted without logging in.

**Review by adamilare — APPROVED** [review]
# STATUS: APPROVED :green_circle: 
## Hi @preciousbetine , 

Your project is complete and ready to merge!  👏 

approved-accepted

Congratulations! 🎉 
HIGHLIGHTS:

- All required changes implemented  :heavy_check_mark: 
- Proper use of gitflow :heavy_check_mark: 
- Proper use of react features :heavy_check_mark: 
- Professional README file  :heavy_check_mark: 
- No linter errors :heavy_check_mark: 

Now you can merge this branch and move on :1st_place_medal: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**